File: 2024/dec01/2.py
#!/usr/bin/python3
from aoc import *
data = list(map(str.split, stdin.readlines()))
a = Counter([int(a[0]) for a in data])
b = Counter([int(b[1]) for b in data])
print(sum([v * c * b[v] for v, c in a.items()]))

File: 2024/dec01/2.pl
#!/usr/bin/perl
use 5.020;
use List::Util qw(sum);
my (@a, %b);
for my $pair (map { [map { int } split / +/] } <>) {
	push @a, $pair->[0];
	++$b{$pair->[1]};
}
say sum(map { abs $_ * $b{$_} } @a);
